### PR TITLE
Add Nodes Labels, Taints and MaxSurge to AgentPool

### PR DESCRIPTION
**Pull request by mjura**
Fixes https://github.com/rancher/rancher/issues/40214

### CODE REVIEW

**Comment by mjura** [comment]
There is missing still part for creating AKS clusters from Rancher UI, only AKS imported clusters were covered

File: pkg/aks/create.go [hunk 1]
```diff
@@ -233,6 +239,14 @@ func CreateOrUpdateAgentPool(ctx context.Context, agentPoolClient services.Agent
 		MinCount:            np.MinCount,
 		MaxCount:            np.MaxCount,
 		VnetSubnetID:        np.VnetSubnetID,
+		NodeLabels:          np.NodeLabels,
+		NodeTaints:          np.NodeTaints,
+	}
+
+	if np.MaxSurge != nil {
```
Comment (alexander-demicev) [inline pkg/aks/create.go:246]: can you also a unit test for this?

File: pkg/aks/create.go [hunk 2]
```diff
@@ -109,8 +109,14 @@ func createManagedCluster(ctx context.Context, cred *Credentials, workplacesClie
 			VMSize:              containerservice.VMSizeTypes(np.VMSize),
 			Mode:                containerservice.AgentPoolMode(np.Mode),
 			OrchestratorVersion: np.OrchestratorVersion,
+			NodeLabels:          np.NodeLabels,
+			NodeTaints:          np.NodeTaints,
+		}
+		if np.MaxSurge != nil {
```
Comment (alexander-demicev) [inline pkg/aks/create.go:115]: can you add a unit test for these fields? Let's make sure we don't lose the functionality in future.

File: pkg/aks/create.go (hunk 1 above)
Comment (mjura) [reply pkg/aks/create.go:246]: I have added unit test

File: pkg/aks/create.go (hunk 2 above)
Comment (mjura) [reply pkg/aks/create.go:115]: I have added unit test

**Review by alexander-demicev — APPROVED** [review]

**Pull request merged**